### PR TITLE
Script running tests using Docker

### PR DESCRIPTION
**Pull request by lucafavatellawork**
PR based on ~#22~ master.

~Maybe the current `sparkl/sse:latest` is not up to date, but I managed to get these tests running using a local Docker image:~ ~make ci_test-via-docker~

The `sparkl/sse:latest` is not up to date so the log tests fail. You can run the tests against a custom image by:
```
env TEST_SSE_DOCKER_IMAGE=sparkl/sse:xyz make ci_test-via-docker
```
You can build a Docker image from https://github.com/opensparkl/sse_core by running `make -C docker docker` (the Docker image tag will have a commit hash).

TODO:
- [x] Make Docker image name configurable from command line.
- [x] Refine target names after #34 settled.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -117,6 +117,31 @@ default-test:
...
+
+.PHONY: default-test-via-docker-setup
+default-test-via-docker-setup:
+	docker run \
+		--name $(test_sse_docker_container) \
+		-d \
+		-it \
+		-p8000:8000 -p9000:9000 \
+		-e CLAIM=postoffice \
+		sparkl/sse:latest
```
Comment (lucafavatellawork) [inline Makefile]: This shall be overridable by the caller.

Comment (lucafavatellawork) [reply Makefile]: Addressed in recent version of the branch.

**Comment by lucafavatellawork** [comment]
I updated the instructions on how to run this PR.
The image `sparkl/sse:latest` is not up to date.

File: Makefile [hunk 2]
```diff
@@ -175,6 +175,32 @@ ci_test-via-local:
...
+.PHONY: ci_test-via-docker-setup
+ci_test-via-docker-setup: export TEST_SSE_DOCKER_IMAGE ?= $(test_sse_docker_image)
+ci_test-via-docker-setup:
+	docker run \
+		--name $(test_sse_docker_container) \
+		-d \
+		-p8000:8000 -p9000:9000 \
+		-e CLAIM=postoffice \
+		$(TEST_SSE_DOCKER_IMAGE)
+	sleep 5
```
Comment (lucafavatellawork) [inline Makefile]: This is hardcoded, I did not find a better way for waiting for the node to start.

**Comment by lucafavatellawork** [comment]
Requires update following recent merge on master.

**Comment by lucafavatellawork** [comment]
In sse_core repo:
```
$ make -C docker docker
...
Successfully tagged sparkl/sse:8f8f519
```

In sse_cli repo, `make deps compile` then:
```
$ env TEST_SSE_DOCKER_IMAGE=sparkl/sse:8f8f519 make ci_test-via-docker
... 53 passed in 11.87s ...
```
... with exit status zero. I ran the same command again, with success.

File: Makefile (hunk 2 above)
Comment (lucafavatellawork) [reply Makefile]: @miklosduma I made this configurable.

File: Makefile [hunk 3]
```diff
@@ -176,6 +176,33 @@ ci_test-via-local:
...
+.PHONY: ci_test-via-docker-setup
+ci_test-via-docker-setup: export TEST_SSE_DOCKER_IMAGE ?= $(test_sse_docker_image)
+ci_test-via-docker-setup: export TEST_SSE_DOCKER_STARTUP ?= $(test_sse_docker_startup)
+ci_test-via-docker-setup:
+	docker run \
+		--name $(test_sse_docker_container) \
+		-d \
+		-p8000:8000 -p9000:9000 \
+		$(TEST_SSE_DOCKER_IMAGE)
+	sleep $(TEST_SSE_DOCKER_STARTUP)
```
Comment (jacobythwaites) [inline Makefile]: Snore... ;)

**Pull request merged**